src/modules/swap/ModuleSwap.vue

```diff
@@ -881,11 +883,13 @@ export default {
      * Set the max available amount to swap from
      */
     setMaxAmount() {
-      this.tokenInValue = this.isFromTokenMain
-        ? new BigNumber(this.availableBalance)
-            .minus(fromWei(MIN_GAS_WEI))
-            .toFixed()
-        : this.availableBalance.toFixed();
+      const availableBalanceMinusGas = new BigNumber(this.availableBalance)
+        .minus(fromWei(MIN_GAS_WEI))
+        .toFixed();
+      this.tokenInValue =
+        this.isFromTokenMain && availableBalanceMinusGas > 0
```

Comment:
So the only reason you want to use availableBalanceMinusGas is when you actually swap main chain currency Like ETH, BNB  and Matic (isFromTokenMain). You need to subtract min gas fee from the balance.
Here you are checking if the (isFromTokenMain and available is greater then 0, if they availableBalanceMinusGas > 0 --> then you get the whole available balance when you should be setting it to 0

Reply:
I think this is fine but `availableBalanceMinusGas > 0` you cant compare bignumbers with `> <` signs you have to use `.gt or .gte`

Reply:
oh @olgakup I see what you mean